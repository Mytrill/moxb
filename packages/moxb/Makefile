all:
	cd ../.. && $(MAKE)

# this will forward any other make goal to the parent makefile
$(MAKECMDGOALS):
	cd ../.. && $(MAKE) $(MAKECMDGOALS)
