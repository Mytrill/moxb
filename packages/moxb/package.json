{
  "name": "@moxb/moxb",
  "version": "0.1.3",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "build",
  "typings": "build/",
  "files": [
    "build/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "dependencies": {
    "tslib": "^1.9.3"
  },
  "peerDependencies": {
    "mobx": "^4.3.0 || ^5.0.0"
  },
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@types/meteor": "^1.4.16",
    "jest": "^23.3.0",
    "prettier": "^1.13.7",
    "ts-jest": "^23.0.0",
    "typescript": "^2.9.2"
  },
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ]
}
