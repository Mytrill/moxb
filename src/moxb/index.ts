export * from './impl';
export * from './meteor';
export * from './types';
