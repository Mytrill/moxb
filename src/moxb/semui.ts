export * from './action/ActionUi';
export * from './bind/BindUi';
export * from './bool/BoolUi';
export * from './confirm/ConfirmUi';
export * from './many-of/ManyOfUi';
export * from './modal/ModalUi';
export * from './numeric/NumericUi';
export * from './one-of/OneOfUi';
export * from './table/TablePaginationUi';
export * from './table/TableSearchUi';
export * from './table/TableUi';
export * from './text/TextUi';
