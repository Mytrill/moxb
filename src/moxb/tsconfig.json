{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["DOM","ES6","ES2015","ES2017", "DOM.Iterable","ScriptHost"],
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "../build",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false
    },
    "compilerOptions": {
        "baseUrl": "",
        "paths": {
            "*": [
                "*",
                "../../optional-modules/node_modules/*"
            ]
        }
    },
    "exclude": [
        "../node_modules"
    ]
}
