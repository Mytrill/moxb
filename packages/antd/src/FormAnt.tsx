import { Form, Alert } from 'antd';
import * as React from 'react';
import { observer } from 'mobx-react';
import { FormProps } from 'antd/lib/form/Form';
import { parseProps } from './BindAnt';
import { Form as MoxForm, t } from '@moxb/moxb';

export interface BindFormAntProps extends React.HTMLProps<HTMLFormElement> {
    operation: MoxForm;
}

@observer
export class FormAnt extends React.Component<FormProps & BindFormAntProps> {
    render() {
        const { operation, children, ...props } = parseProps(this.props, this.props.operation);
        return (
            <Form onSubmit={operation.onSubmitForm} {...props as any}>
                {children}
                {operation.hasErrors && (
                    <Alert
                        message={t('FormAnt.errors.header', 'Errors')}
                        description={operation.allErrors}
                        type="error"
                        closable
                        onClose={operation.clearErrors}
                    />
                )}
            </Form>
        );
    }
}
