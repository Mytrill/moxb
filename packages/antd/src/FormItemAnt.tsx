import { Form } from 'antd';
import { ColProps } from 'antd/lib/grid';
import { CSSProperties } from 'react';
import * as React from 'react';
import { observer } from 'mobx-react';
import { parseProps, labelWithHelp, getErrorMessages } from './BindAnt';
import { FormItem as MoxFormItem } from '@moxb/moxb';
import { FormItemProps } from 'antd/lib/form/FormItem';

export interface BindFormItemAntProps extends React.HTMLProps<HTMLFormElement> {
    operation: MoxFormItem;
    formStyle?: CSSProperties;
    wrapperCol?: ColProps;
    labelCol?: ColProps;
    label?: string;
}

@observer
export class FormItemAnt extends React.Component<FormItemProps & BindFormItemAntProps> {
    render() {
        const { operation, children, label, formStyle, labelCol, wrapperCol, invisible, ...props } = parseProps(
            this.props,
            this.props.operation
        );
        if (invisible) {
            return null;
        }
        return (
            <Form.Item
                label={labelWithHelp(label != null ? label : operation.label, operation.help)}
                style={formStyle || undefined}
                labelCol={labelCol}
                wrapperCol={wrapperCol}
                required={operation.required}
                hasFeedback={operation.hasErrors}
                validateStatus={operation.hasErrors ? 'error' : undefined}
                help={operation.hasErrors ? getErrorMessages(operation.errors!) : null}
                {...props as any}
            >
                {children}
            </Form.Item>
        );
    }
}
