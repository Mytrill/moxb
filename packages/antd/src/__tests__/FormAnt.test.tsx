import { FormImpl, TextImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { FormAnt } from '../FormAnt';
import { FormItemAnt } from '../FormItemAnt';

describe('FormItemAnt', function() {
    const formUserText = new TextImpl({
        id: 'id.formUserText',
        label: 'Username',
        onSave: (bind, done) => jest.fn(),
    });
    const formPasswordText = new TextImpl({
        id: 'id.formUserText',
        label: 'Username',
        onSave: (bind, done) => jest.fn(),
    });
    function newFormOperation(opt?: any) {
        const onSubmitMock = jest.fn();
        return new FormImpl(
            Object.assign(
                {
                    id: 'id.testForm',
                    values: [formUserText, formPasswordText],
                    onSubmit: (bind: any, done: any) => onSubmitMock,
                },
                ...opt
            )
        );
    }

    it('should render a form element by default', function() {
        const operation = newFormOperation();
        expect(shallow(<FormAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should throw if no values for children were defined', function() {
        const onSubmitMock = jest.fn();
        expect(() => {
            new FormImpl({
                id: 'id.testForm',
                values: [],
                onSubmit: (bind, done) => onSubmitMock,
            });
        }).toThrow();
    });

    it('should return null if invisible', function() {
        const operation = newFormOperation({ invisible: () => true });
        expect(shallow(<FormAnt operation={operation} />).type()).toBeNull();
    });

    it('should show an alert, if an error happens in a child component', function() {
        const operation = newFormOperation();
        formUserText.setError('New error');
        const wrapper = shallow(<FormAnt operation={operation} />);
        expect(wrapper.render().find('div.ant-alert').length).toBe(1);
    });

    it('should not show an alert on error with property "hideErrors" set', function() {
        const operation = newFormOperation();
        formUserText.setError('New error');
        const wrapper = shallow(<FormAnt hideErrors operation={operation} />);
        expect(wrapper.render().find('div.ant-alert').length).toBe(0);
    });

    it('should show an alert, if an error in the form happened', function() {
        const operation = newFormOperation();
        operation.setError('New error');
        const wrapper = shallow(<FormAnt operation={operation} />);
        expect(wrapper.render().find('div.ant-alert').length).toBe(1);
    });
});
