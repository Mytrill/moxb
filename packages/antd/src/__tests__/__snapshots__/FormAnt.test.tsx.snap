// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItemAnt should render a form element by default 1`] = `
<Form
  disabled={false}
  hideRequiredMark={false}
  id="id-test_form"
  layout="horizontal"
  onSubmit={[Function]}
  prefixCls="ant-form"
  readOnly={false}
/>
`;
